src/main.rs

```diff
@@ -160,36 +163,27 @@ macro_rules! route_request {
 #[rtic::app(device = stm32h7xx_hal::stm32, peripherals = true, monotonic = rtic::cyccnt::CYCCNT)]
 const APP: () = {
     struct Resources {
-        adc0: hal::spi::Spi<hal::stm32::SPI2>,
-        dac0: hal::spi::Spi<hal::stm32::SPI4>,
+        adc0: hal::spi::Spi<hal::stm32::SPI2, hal::spi::Enabled, u16>,
+        dac0: hal::spi::Spi<hal::stm32::SPI4, hal::spi::Enabled, u16>,
         afe0: AFE0,
 
-        adc1: hal::spi::Spi<hal::stm32::SPI3>,
-        dac1: hal::spi::Spi<hal::stm32::SPI5>,
+        adc1: hal::spi::Spi<hal::stm32::SPI3, hal::spi::Enabled, u16>,
+        dac1: hal::spi::Spi<hal::stm32::SPI5, hal::spi::Enabled, u16>,
         afe1: AFE1,
 
         eeprom_i2c: hal::i2c::I2c<hal::stm32::I2C2>,
 
         timer: hal::timer::Timer<hal::stm32::TIM2>,
 
         // Note: It appears that rustfmt generates a format that GDB cannot recognize, which
```

Comment:
This is actually a rustc bug - in certain situations it generates incorrect debug information in the ELF and GDB can't parse it (because it's wrong). There's a rustc issue filed, but I don't remember exactly where anymore.